### ISSUE TITLE
[ios_xctestrun_runner] codesigning/entitlements blockers

### ISSUE THREAD

**Issue by iainsmith** [bazelbuild/rules_apple#1912]
## Summary

ios_xctestrun_runner doesn't match codesigning/entitlements from Xcodes XCTRunner.app, which blocks certain API's (in our case a Network.framework based HTTP Server). [This Apple forums post](https://developer.apple.com/forums/thread/114907), seems to indicate this as the likely cause.

- [ios_xctestrun_runner code](https://github.com/bazelbuild/rules_apple/blob/ad91a19f770a43aeb948ffcc8c8f915abda6a6a2/apple/testing/default_runner/ios_xctestrun_runner.template.sh#L147-L151), copies the supporting frameworks directly
- [google/xctestrunner code](https://github.com/google/xctestrunner/blob/24629f3e6c0dda397f14924b64eb45d04433c07e/xctestrunner/test_runner/xctestrun.py#L454-L480), codesigns and matches XCTRunner.app/RunnerEntitlements.plist.

## More details

We tried migrating to the new ios_xctestrunner for our UITests and hit this issue. We did a bit of digging, and think it's related to the codesigning/entitlements. Our UITest try to start a (Network.framework) based MockServer, which fails to launch.

There is a previous [issue #504](https://github.com/bazelbuild/rules_apple/issues/504) about moving the codesigning responsibilities outside of the test runner from a couple of years ago.

I was wondering if the test runner can call `tools/codesigningtool.py` or if it's better to just shell out to `codesign -f` directly in the `ios_xctestrun_runner.template.sh`

cc @mattrobmattrob

**Comment by keith** [comment]
@maxwellE

**Comment by maxwellE** [comment]
So I assume that you are running on device? Or even using Network.framework on sim crashes?

Feel free to assign to me. Ideally we have a test case for this scenario as well

**Comment by iainsmith** [comment]
> So I assume that you are running on device? Or even using Network.framework on sim crashes?

It happens on the simulator. 

We probably crash because we assume that server always starts. I think the NWListener can't bind to the port on the simulator using the new runner.

I will try and post a minimal reproducer later.

**Comment by tinder-maxwellelliott** [comment]
@Keith you can assign to me

**Comment by maxwellE** [comment]
@keith you can assign to me, should be able to take a look today

**Comment by maxwellE** [comment]
@iainsmith Can you check out https://github.com/bazelbuild/rules_apple/pull/1932 and see if that works for you?

**Comment by iainsmith** [comment]
Hey @maxwellE I tried this out locally, but I'm hitting the same error where we can't start the NWListener.

I can see these lines in the log,

```
entitlements_path=/var/folders/9j/_v7frzl550x84_qn_8d6rrm00000gq/T//test_tmp_dir.D7dg5F/XCTRunner.app/RunnerEntitlements.plist
find /var/folders/9j/_v7frzl550x84_qn_8d6rrm00000gq/T//test_tmp_dir.D7dg5F/XCTRunner.app/Frameworks -name '*.framework' -exec codesign -f --entitlements /var/folders/9j/_v7frzl550x84_qn_8d6rrm00000gq/T//test_tmp_dir.D7dg5F/XCTRunner.app/RunnerEntitlements.plist --timestamp=none -s - '{}' ';'
codesign -f --entitlements /var/folders/9j/_v7frzl550x84_qn_8d6rrm00000gq/T//test_tmp_dir.D7dg5F/XCTRunner.app/RunnerEntitlements.plist --timestamp=none -s - /var/folders/9j/_v7frzl550x84_qn_8d6rrm00000gq/T//test_tmp_dir.D7dg5F/XCTRunner.app

Unable to start mock server on port: 11704 (some random port)
```

**Comment by maxwellE** [comment]
@iainsmith Can you push up a small repro to this branch for me to test with?

**Comment by maxwellE** [comment]
I am starting to investigate this again

**Comment by iainsmith** [comment]
Fixed in 2.4.0
